### ISSUE TITLE
The latent representation of compiler phase order

### ISSUE THREAD

**Issue by gloaming2dawn** [yangkevin2/plalam#5]
Hi, thank you for your good work. 

Can I know how do you learn the latent representation of compiler phase order task? That is not mentioned in the paper.

**Comment by yangkevin2** [comment]
@tianjunz

**Comment by tianjunz** [comment]
We used a random representation (e.g., projecting the state space using a random MLP network). It is very similar to the maze environment except changing the CNN network into a MLP one.

**Comment by gloaming2dawn** [comment]
> We used a random representation (e.g., projecting the state space using a random MLP network). It is very similar to the maze environment except changing the CNN network into a MLP one.

Thanks for your reply. Several other questions:

Did you use a pretrained network, or just use the collected label during search to train the MLP network?

Is the network an auto-encoder and you use a decoder to obtain the candidate? if so, how do you train it? For example, jointly training a predictor and auto-encoder-decoder?

@tianjunz

**Comment by tianjunz** [comment]
No, we don't pretrain/train the network. We use a random network for projecting. So pretty much a randomly initialized MLP could work. We use this for classification but not for sampling. You can achieve this by setting "--latent=True" and "--latent_sample=False".

**Comment by gloaming2dawn** [comment]
Hi, I think I still have a new question, i.e.， what is the state space for compiler phase order problem? Is it the action sequence or the binary file?
